### PR TITLE
Adding changes for Language Authoring 2023-04-01 to dev folder

### PR DESCRIPTION
**Pull request by AmirGeorge**
# Data Plane API - Pull Request
This PR is adding changes for Language Authoring 2023-04-01 to dev folder. The GA 2023-04-01 APIs will ship only some of the features we surfaced in 2022-10-01-preview. So since the dev folder on main currently contains the swaggers for 2022-10-01-preview, the diff here is removing the features that we won't be surfacing. Nothing introduced in the GA 2022-05-01 is removed here.

Is this review for (select one):

- [ ] a private preview
- [ ] a public preview
- [✅ ] GA release

### CODE REVIEW

**Comment by heaths** [comment]
Can you describe what these two similar PRs are for? If you're preparing to release, you should create a release branch from main - nor a PR targeting main. As it is, you have two PRs now competing to target main when neither should.

In this case, since the GA will happen first, it makes sense to create a stabilization (release) branch, finish up work, and snapshot to the specification/ folder. Work on the preview can continue in main. When the GA is released, merge this into main (or not - maybe nothing changed that needs to be merged, or just cherry-pick what commits you need) and then we repeat the process for the preview (stabilization branch, snapshot to specification/ folder). This is explained in the [walkthrough](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/699/Walkthrough-Creating-a-new-service-specification).

/cc @kayousef @mayurid @kristapratico

**Comment by AmirGeorge** [comment]
I reviewed the breaking changes and almost all of them are expected.  Nothing here is removing any of our GA features. The state is that we GA'ed some features in 2022-05-01 APIs, then we shipped 2022-01-10-preview with extra preview features added. The state now is that in the upcoming GA version 2023-04-01, we will not be GAing all the preview features in 2022-10-01-preview, just a subset of them. So since the dev folder on main branch currently contains 2022-10-01-preview, the PR is showing removals reflecting not GAing those preview features in in 2023-04-01. But all the GA features in 2022-05-01 will be included in the upcoming GA version 2023-04-01, as well as a subset of the 2022-10-01-preview features.

For the rest of the breaking changes, I will be leaving questions here to double check that nothing weird is happening.

**Review by kristapratico — APPROVED** [review]

File: dev/cognitiveservices/data-plane/Language/analyzeconversations-authoring.json
Comment (heaths) [inline dev/cognitiveservices/data-plane/Language/analyzeconversations-authoring.json:1]: Looking at the [diff from the last GA](https://portal.azure-devex-tools.com/tools/diff/eyJzb3VyY2VVcmwiOiJodHRwczovL3Jhdy5naXRodWJ1c2VyY29udGVudC5jb20vQXp1cmUvYXp1cmUtcmVzdC1hcGktc3BlY3MvbWFpbi9zcGVjaWZpY2F0aW9uL2NvZ25pdGl2ZXNlcnZpY2VzL2RhdGEtcGxhbmUvTGFuZ3VhZ2Uvc3RhYmxlLzIwMjItMDUtMDEvYW5hbHl6ZWNvbnZlcnNhdGlvbnMtYXV0aG9yaW5nLmpzb24iLCJ0YXJnZXRVcmwiOiJodHRwczovL3Jhdy5naXRodWJ1c2VyY29udGVudC5jb20vQXp1cmUvYXp1cmUtcmVzdC1hcGktc3BlY3MvNDEwYjBhNWExMDhjNmFlYjJmOTI1N2I4OTEzNzY4YzkyMjM1M2VlMi9kZXYvY29nbml0aXZlc2VydmljZXMvZGF0YS1wbGFuZS9MYW5ndWFnZS9hbmFseXplY29udmVyc2F0aW9ucy1hdXRob3JpbmcuanNvbiJ9), I don't see anything concerning except for the breaking change of adding a required field below. See https://aka.ms/AzBreakingChangesPolicy for details.

File: dev/cognitiveservices/data-plane/Language/analyzetext-authoring.json [hunk 1]
```diff
@@ -3209,7 +2590,8 @@
         "lastTrainedDateTime",
         "lastTrainingDurationInSeconds",
         "modelExpirationDate",
-        "modelTrainingConfigVersion"
+        "modelTrainingConfigVersion",
+        "hasSnapshot"
```
Comment (heaths) [inline dev/cognitiveservices/data-plane/Language/analyzetext-authoring.json:2594]: Adding a required property is a breaking change.

**Review by heaths — CHANGES_REQUESTED** [review]
The [diff from the last GA](https://portal.azure-devex-tools.com/tools/diff/eyJzb3VyY2VVcmwiOiJodHRwczovL3Jhdy5naXRodWJ1c2VyY29udGVudC5jb20vQXp1cmUvYXp1cmUtcmVzdC1hcGktc3BlY3MvbWFpbi9zcGVjaWZpY2F0aW9uL2NvZ25pdGl2ZXNlcnZpY2VzL2RhdGEtcGxhbmUvTGFuZ3VhZ2Uvc3RhYmxlLzIwMjItMDUtMDEvYW5hbHl6ZXRleHQtYXV0aG9yaW5nLmpzb24iLCJ0YXJnZXRVcmwiOiJodHRwczovL3Jhdy5naXRodWJ1c2VyY29udGVudC5jb20vQXp1cmUvYXp1cmUtcmVzdC1hcGktc3BlY3MvNDEwYjBhNWExMDhjNmFlYjJmOTI1N2I4OTEzNzY4YzkyMjM1M2VlMi9kZXYvY29nbml0aXZlc2VydmljZXMvZGF0YS1wbGFuZS9MYW5ndWFnZS9hbmFseXpldGV4dC1hdXRob3JpbmcuanNvbiJ9) looks fine except for the newly added required property, which is a breaking change.

File: dev/cognitiveservices/data-plane/Language/analyzeconversations-authoring.json [hunk 2]
```diff
@@ -3328,7 +2714,8 @@
         "lastTrainedDateTime",
         "lastTrainingDurationInSeconds",
         "modelExpirationDate",
-        "modelTrainingConfigVersion"
+        "modelTrainingConfigVersion",
+        "hasSnapshot"
```
[the remark this answers is not in the document]
Comment (heaths) [reply dev/cognitiveservices/data-plane/Language/analyzeconversations-authoring.json:2718]: Actually, I see this is a response-only model so adding a required property only requires a new api-version, which you're doing. Sorry.

File: dev/cognitiveservices/data-plane/Language/analyzetext-authoring.json (hunk 1 above)
Comment (heaths) [reply dev/cognitiveservices/data-plane/Language/analyzetext-authoring.json:2594]: Actually, I see this is a response-only model so adding a required property only requires a new api-version, which you're doing. Sorry.

**Review by heaths — APPROVED** [review]

**Pull request merged**